src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -298,29 +391,46 @@ private static Solution GetSolutionWithFormattedInterfaceDocument(Document unfor
             // extracted members are also not updated.
             if (typeToExtractFrom.TypeKind == TypeKind.Interface)
             {
-                return solutionWithFormattedInterfaceDocument;
+                return solution;
             }
 
-            var formattedSolution = GetSolutionWithUpdatedOriginalType(
-                solutionWithFormattedInterfaceDocument,
+            var unformattedSolution = solution;
+            foreach (var documentId in documentIds)
+            {
+                var document = solution.GetDocument(documentId);
+                var currentRoot = document.GetSyntaxRootSynchronously(cancellationToken);
+                var editor = new SyntaxEditor(currentRoot, solution.Workspace);
+
+                var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+                var typeReference = syntaxGenerator.TypeExpression(extractedInterfaceSymbol);
+
+                var typeDeclaration = currentRoot.GetAnnotatedNodes(typeNodeAnnotation).SingleOrDefault();
```

Comment:
why this one use SingleOrDefault? when the other one uses just Single?

Reply:
Yes. It's possible the document doesn't, which is fine. We just want to modify the first we come across.